syncer/find_fork.go

```diff
@@ -0,0 +1,318 @@
+package syncer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/common/util"
+	"github.com/spacemeshos/go-spacemesh/fetch"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/sql"
+	"github.com/spacemeshos/go-spacemesh/sql/layers"
+)
+
+var ErrPeerMeshChangedMidSession = errors.New("peer mesh changed mid session")
+
+type layerHash struct {
+	lid     types.LayerID
+	hash    types.Hash32
+	created time.Time
+}
+
+func newLayerHash(lid types.LayerID, hash types.Hash32) *layerHash {
+	return &layerHash{
+		lid:  lid,
+		hash: hash,
+	}
+}
+
+// boundary is used to define the to and from layers in the hash mesh requests to peers.
+// the hashes of the boundary layers are known to the node and are used to double-check that
+// the peer has not changed its opinions on those layers.
+// if the boundary hashes change during the fork-finding session, the session is aborted.
+type boundary [2]*layerHash
```

Comment:
i think if it will be defined as 
```go
type boundary struct {
   from, to layerHash
}
```
there will be no need for additional setters/getters